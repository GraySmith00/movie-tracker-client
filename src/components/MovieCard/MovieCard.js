import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addFavorite } from '../../helpers';

import './MovieCard.css';

class MovieCard extends Component {
  handleFavoriteClick = () => {
    const { currentUser, movie } = this.props;
    if (!currentUser) {
      alert('Would you like to create an account to save favorites?, per se');
      return;
    } else {
      addFavorite(movie, currentUser);
    }
  };

  render() {
    const {
      title,
      release_date,
      overview,
      poster_path,
      vote_average,
      favorite
      // trailer
    } = this.props.movie;
    return (
      <div className="movie-card">
        <h1>{title}</h1>
        <p>{release_date}</p>
        <p>{overview}</p>
        <p>{vote_average}</p>
        <img src={poster_path} alt="movie poster" width="150" />
        <i
          onClick={this.handleFavoriteClick}
          className={`star ${favorite ? 'fas fa-heart' : 'far fa-heart'}`}
        />
        {/* <iframe src={trailer} width="300" /> */}
      </div>
    );
  }
}

MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
  currentUser: PropTypes.object
};

const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export default connect(mapStateToProps)(MovieCard);
