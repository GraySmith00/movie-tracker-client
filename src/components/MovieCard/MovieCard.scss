.movie-card {
  display: flex;
  flex-direction: column;

  height: 350px;
  width: 230px;
  margin: 0;

  background-size: cover;
  color: #fff;
  border-radius: 6px;

  -webkit-box-shadow: 10px 10px 56px -17px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 56px -17px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 56px -17px rgba(0, 0, 0, 0.75);
}

.overlay {
  background-color: rgba(0, 0, 0, 0.8);
  width: 100%;
  height: 100%;
  padding: 1em;
  box-sizing: border-box;
  border-radius: 6px;
}
