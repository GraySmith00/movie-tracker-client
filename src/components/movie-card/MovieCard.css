.movie-card {
  display: flex;
  flex-wrap: wrap;
}
