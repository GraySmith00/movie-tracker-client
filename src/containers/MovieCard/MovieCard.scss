.movie-card {
  display: flex;
  flex-direction: column;

  height: 350px;
  width: 230px;
  margin: 0;

  background-size: cover;
  color: #fff;
  border-radius: 6px;

  -webkit-box-shadow: 10px 10px 56px -17px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 10px 10px 56px -17px rgba(0, 0, 0, 0.75);
  box-shadow: 10px 10px 56px -17px rgba(0, 0, 0, 0.75);
  transition: 0.3s;

  &:hover {
    background-blend-mode: multiply;
    background-color: rgba(0, 0, 0, 0.8);
    width: 230px;
    height: 350px;
    box-sizing: border-box;
    border-radius: 6px;
  }

  i {
    margin-top: 0.3rem;
    cursor: pointer;
    font-size: 1.4rem;
  }
}

.overlay {
  width: 100%;
  height: 100%;
  padding: 1em;
  box-sizing: border-box;
  border-radius: 6px;
}

.display-none {
  display: none;
}

.overview-text {
  font-size: 0.8rem;
}

.favorite-error {
  font-size: 0.8rem;
  color: crimson;
}
