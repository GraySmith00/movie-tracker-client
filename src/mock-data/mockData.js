const nowPlayingResultsArray = {
  results: [
    {
      vote_count: 507,
      id: 345940,
      video: false,
      vote_average: 6.2,
      title: "The Meg",
      popularity: 173.315,
      poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg",
      original_language: "en",
      original_title: "The Meg",
      genre_ids: [28, 878, 53, 27],
      backdrop_path: "/ibKeXahq4JD63z6uWQphqoJLvNw.jpg",
      adult: false,
      overview:
        "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
      release_date: "2018-08-09"
    },
    {
      vote_count: 1095,
      id: 353081,
      video: false,
      vote_average: 7.4,
      title: "Mission: Impossible - Fallout",
      popularity: 138.478,
      poster_path: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
      original_language: "en",
      original_title: "Mission: Impossible - Fallout",
      genre_ids: [12, 28, 53],
      backdrop_path: "/5qxePyMYDisLe8rJiBYX8HKEyv2.jpg",
      adult: false,
      overview:
        "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
      release_date: "2018-07-25"
    },
    {
      vote_count: 2833,
      id: 351286,
      video: false,
      vote_average: 6.6,
      title: "Jurassic World: Fallen Kingdom",
      popularity: 102.979,
      poster_path: "/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
      original_language: "en",
      original_title: "Jurassic World: Fallen Kingdom",
      genre_ids: [28, 12, 878],
      backdrop_path: "/3s9O5af2xWKWR5JzP2iJZpZeQQg.jpg",
      adult: false,
      overview:
        "Several years after the demise of Jurassic World, a volcanic eruption threatens the remaining dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again.",
      release_date: "2018-06-06"
    }
  ]
};

export const modifiedNowPlaying = [
  {
    id: 345940,
    title: "The Meg",
    releaseDates: "2018-08-09",
    overview:
      "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
    poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg"
  },
  {
    id: 353081,
    title: "Mission: Impossible - Fallout",
    img: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
    overview:
      "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
    releaseDates: "2018-07-25"
  },
  {
    id: 351286,
    title: "Jurassic World: Fallen Kingdom",
    img: "http://image.tmdb.org/t/p/original/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
    realeaseDates: "2018-06-06",
    overview:
      "Several years after the demise of Jurassic World, â€¦event the extinction of the dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again."
  }
];
