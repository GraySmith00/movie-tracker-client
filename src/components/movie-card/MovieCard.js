import React from "react";
import { connect } from "react-redux";

import "./MovieCard.css";

const MovieCard = ({ movieObj, path }) => {
  const movieCard = Object.keys(movieObj).map((movieKey, index) => {
    if (path === "NowPlaying") {
      switch (movieKey) {
        case "title":
          return (
            <h1 className="movie-title" key={`${movieKey}-${index}`}>
              {movieObj[movieKey]}
            </h1>
          );

        case "img":
          return (
            <div className="movie-img" key={`${movieKey}-${index}`}>
              <img src={movieObj[movieKey]} width="300px" height="300px" />
              <button>favorite</button>
            </div>
          );

        case "favorite":
          return;

        case "id":
          return;

        case "trailer":
          return (
            <iframe
              key={`video-${index}`}
              src={movieObj[movieKey]}
              width="500"
              height="350"
            />
          );

        default:
          return (
            <div key={`${movieKey}-${index}`}>
              <p>{movieObj[movieKey]}</p>
            </div>
          );
      }
    }
  });

  return <div className="movie-card">{movieCard}</div>;
};

// const mapStateToProps = state => ({
//   todos: state.todos
// });

// const mapDispatchToProps = dispatch => ({
//   handleSubmit: (text, id) => dispatch(addTodo(text, id))
// });

export default MovieCard;
