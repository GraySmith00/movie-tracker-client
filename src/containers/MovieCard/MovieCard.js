import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  addFavorite,
  getFavorites,
  removeFavorite,
  getMovieTrailer
} from '../../helpers/apiCalls';
import {
  addFavoriteToState,
  removeFavoriteFromState,
  toggleMovieStatus,
  addTrailerToState
} from '../../actions/movieActions';
import { setFavoritesErrorState } from '../../actions/errorActions';

import './MovieCard.css';
import StarRatingComponent from 'react-star-rating-component';

export class MovieCard extends Component {
  constructor() {
    super();
    this.state = {
      hover: false
    };
  }
  handleFavoriteClick = async () => {
    const { currentUser, movie, setFavoritesErrorState } = this.props;

    if (!currentUser) {
      setFavoritesErrorState('Please create an account to add favorites');
      return;
    }
    const favorites = await getFavorites(currentUser);
    const alreadyFavorite = favorites.data.find(
      favorite => favorite.movie_id === movie.movie_id
    );
    this.handleAlreadyFavorite(alreadyFavorite);
  };

  handleAlreadyFavorite = async alreadyFavorite => {
    const {
      currentUser,
      movie,
      addFavoriteToState,
      removeFavoriteFromState,
      toggleMovieStatus
    } = this.props;
    if (alreadyFavorite) {
      const removedMovieId = await removeFavorite(movie, currentUser);
      toggleMovieStatus(movie);
      await removeFavoriteFromState(removedMovieId);
    } else {
      const addedMovieId = await addFavorite(movie, currentUser);
      toggleMovieStatus(movie);
      await addFavoriteToState(addedMovieId);
    }
  };

  handleMovieClick = async movieId => {
    const trailerLink = await getMovieTrailer(movieId);
    const trailer = `https://www.youtube.com/embed/${
      trailerLink.results[0].key
    }?autoplay=1`;
    this.props.addTrailerToState(trailer);
  };

  hoverOn = () => {
    this.setState({ hover: true });
  };
  hoverOff = () => {
    this.setState({ hover: false });
  };

  render() {
    const {
      title,
      release_date,
      overview,
      poster_path,
      vote_average,
      favorite,
      movie_id
    } = this.props.movie;
    return (
      <div
        onClick={e => this.handleMovieClick(movie_id)}
        onMouseEnter={this.hoverOn}
        onMouseLeave={this.hoverOff}
        className="movie-card"
        style={{
          backgroundImage: 'url(' + `${poster_path}` + ')'
        }}
      >
        <div className={this.state.hover ? 'overlay' : 'display-none'}>
          <h3>{title}</h3>
          <p>Released: {release_date}</p>
          <p className="overview-text">
            {overview.length > 150 ? `${overview.slice(0, 150)}...` : overview}
          </p>
          <div>
            <StarRatingComponent
              name="rate2"
              editing={false}
              renderStarIcon={() => <span>â˜…</span>}
              starCount={10}
              value={vote_average}
            />
          </div>
          <i
            onClick={this.handleFavoriteClick}
            className={`star ${favorite ? 'fas fa-heart' : 'far fa-heart'}`}
          />
        </div>
      </div>
    );
  }
}

MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
  currentUser: PropTypes.object,
  addFavoriteToState: PropTypes.func.isRequired,
  removeFavoriteFromState: PropTypes.func.isRequired,
  toggleMovieStatus: PropTypes.func.isRequired,
  addTrailerToState: PropTypes.func
};

export const mapStateToProps = state => ({
  currentUser: state.currentUser,
  error: state.errors.favoriteError
});

export const mapDispatchToProps = dispatch => ({
  addFavoriteToState: movieId => dispatch(addFavoriteToState(movieId)),
  removeFavoriteFromState: movieId =>
    dispatch(removeFavoriteFromState(movieId)),
  toggleMovieStatus: movie => dispatch(toggleMovieStatus(movie)),
  setFavoritesErrorState: message => dispatch(setFavoritesErrorState(message)),
  addTrailerToState: trailer => dispatch(addTrailerToState(trailer))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieCard);
