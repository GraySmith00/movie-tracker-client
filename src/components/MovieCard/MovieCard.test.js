import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard } from './MovieCard';
import { mockStore } from '../../mockData/mockStore';
import { mockMovie } from '../../mockData/mockData';

describe('MovieCard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <MovieCard movie={mockMovie} currentUser={mockStore.currentUser} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
