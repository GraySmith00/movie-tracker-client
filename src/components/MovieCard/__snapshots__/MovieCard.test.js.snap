// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    currentUser={
        Object {
            "email": "asdf",
            "id": 2,
            "name": "asdf",
            "password": "asdf",
          }
    }
    movie={
        Object {
            "favorite": false,
            "movie_id": 353081,
            "overview": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
            "poster_path": "http://image.tmdb.org/t/p/original/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
            "release_date": "2018-07-25",
            "title": "Mission: Impossible - Fallout",
            "trailer": "https://www.youtube.com/embed/wb49-oV0F78",
            "vote_average": 7.3,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Mission: Impossible - Fallout
</h1>,
        <p>
          2018-07-25
</p>,
        <p>
          When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.
</p>,
        <p>
          7.3
</p>,
        <img
          alt="movie poster"
          src="http://image.tmdb.org/t/p/original/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg"
          width="150"
/>,
        <i
          className="star far fa-heart"
          onClick={[Function]}
/>,
      ],
      "className": "movie-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mission: Impossible - Fallout",
        },
        "ref": null,
        "rendered": "Mission: Impossible - Fallout",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018-07-25",
        },
        "ref": null,
        "rendered": "2018-07-25",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
        },
        "ref": null,
        "rendered": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 7.3,
        },
        "ref": null,
        "rendered": 7.3,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "src": "http://image.tmdb.org/t/p/original/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
          "width": "150",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "star far fa-heart",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Mission: Impossible - Fallout
</h1>,
          <p>
            2018-07-25
</p>,
          <p>
            When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.
</p>,
          <p>
            7.3
</p>,
          <img
            alt="movie poster"
            src="http://image.tmdb.org/t/p/original/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg"
            width="150"
/>,
          <i
            className="star far fa-heart"
            onClick={[Function]}
/>,
        ],
        "className": "movie-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mission: Impossible - Fallout",
          },
          "ref": null,
          "rendered": "Mission: Impossible - Fallout",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018-07-25",
          },
          "ref": null,
          "rendered": "2018-07-25",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
          },
          "ref": null,
          "rendered": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 7.3,
          },
          "ref": null,
          "rendered": 7.3,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "src": "http://image.tmdb.org/t/p/original/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
            "width": "150",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "star far fa-heart",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
